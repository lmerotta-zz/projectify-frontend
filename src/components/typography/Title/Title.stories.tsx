// Button.stories.tsx
import { ComponentProps } from "react";
import Title from "./Title";
import { Story } from "@storybook/react/types-6-0";

// eslint-disable-next-line import/no-anonymous-default-export
export default {
  title: "Typography/Title",
  component: Title,
};

const Template: Story<ComponentProps<typeof Title>> = () => (
  <Title>Login</Title>
);

export const Default = Template.bind({});
