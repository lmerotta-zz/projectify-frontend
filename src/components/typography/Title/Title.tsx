import tw from "twin.macro";

export default tw.h1`text-4xl md:text-5xl text-dark font-bold`;
