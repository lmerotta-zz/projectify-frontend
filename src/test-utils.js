import React from "react";
import { render } from "@testing-library/react";
import { CacheProvider } from "@emotion/react";
import createCache from "@emotion/cache";

const cache = createCache({ key: "iodajdoiwanoawdni" });
cache.compat = true;

const AllTheProviders = ({ children }) => {
  return <CacheProvider value={cache}>{children}</CacheProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
