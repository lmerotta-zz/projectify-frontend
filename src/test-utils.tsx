import { MockedProvider, MockedProviderProps } from "@apollo/client/testing";
import {
  CircularProgress,
  StyledEngineProvider,
  ThemeProvider,
} from "@mui/material";
import { render as realRender, RenderOptions } from "@testing-library/react";
import { ReactElement, ReactNode, Suspense } from "react";
import { MemoryRouter, MemoryRouterProps } from "react-router";
import { Route, Routes } from "react-router-dom";
import { ToastContainer } from "react-toastify";
import theme from "theme";

function renderWithProviders(
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries"> & {
    routerProps?: MemoryRouterProps;
    graphqlProps?: MockedProviderProps;
  }
) {
  const Wrapper = ({ children }: { children?: ReactNode }) => {
    return (
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={theme}>
          <ToastContainer
            theme="colored"
            autoClose={3000}
            containerId="react-toastify"
          />
          <MockedProvider {...options?.graphqlProps} addTypename={false}>
            <MemoryRouter {...options?.routerProps}>
              <Suspense fallback={<CircularProgress />}>
                <Routes>
                  <Route
                    path="/*"
                    element={
                      <>
                        <h1>TESTSASATATAST</h1>
                        {children}
                      </>
                    }
                  />
                </Routes>
              </Suspense>
            </MemoryRouter>
          </MockedProvider>
        </ThemeProvider>
      </StyledEngineProvider>
    );
  };

  return realRender(ui, { wrapper: Wrapper, ...options });
}

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
export { renderWithProviders };
