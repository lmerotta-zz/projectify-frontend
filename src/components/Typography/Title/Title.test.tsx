import { render } from "test-utils";
import Title from "./Title";

describe("Title unit tests", () => {
  it("Renders", () => {
    const { asFragment } = render(<Title>My title</Title>);

    expect(asFragment()).toMatchSnapshot();
  });
});
