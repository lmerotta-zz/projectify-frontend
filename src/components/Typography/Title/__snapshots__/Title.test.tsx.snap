// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title unit tests Renders 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  --tw-text-opacity: 1;
  color: rgba(63, 63, 70, var(--tw-text-opacity));
  font-weight: 700;
}

@media (min-width: 768px) {
  .emotion-0 {
    font-size: 3rem;
    line-height: 1;
  }
}

<h1
    class="emotion-0"
  >
    My title
  </h1>
</DocumentFragment>
`;
